tutorials/2_sprite_animations_gestures/README.md

```diff
@@ -0,0 +1,241 @@
+      srcSize: Vector2(60, 20),
+    );
+
+    pressedButton = await loadSprite(
+      'buttons.png',
+      // Same thing here, but now a rect starting at (0, 20)
+      // which gives us only the second button
+      srcPosition: Vector2(0, 20),
+      srcSize: Vector2(60, 20),
+    );
+  }
+```
+
+Finally, we just render it on the game `render` function:
+
+```dart
+  @override
+  void render(Canvas canvas) {
+    // Running robot render omited
+
+    if (isPressed) {
+      pressedButton.render(canvas, position: buttonPosition, size: buttonSize);
+    } else {
+      unpressedButton.render(canvas,
+          position: buttonPosition, size: buttonSize);
+    }
+  }
+```
+
+You now should see the button on the screen, but right now, it is pretty much useless as it has no action at all.
+
+So, to change that, we will now add some interactivity to our game and make the button tapable/clickable.
+
+Flame provides several input handlers, which you can check with more in depth on the [input documentation of Flame](https://github.com/flame-engine/flame/blob/main/doc/input.md). For this example we will be using the `TapDetector` which enables us to detect taps on the screen, as well as mouse click when running on web or desktop.
+
+All of Flame input detectors are mixins which can be added to your game, which when applied, enables you to override listener methods related to that detector. In our example, we will need to override three methods:
+
+ - `onTapDown`: Called when touch/click has started. i.e. the user just touced the screen or clicked the mouse button.
+ - `onTapUp`: Called when the touch/click has stop ocurring because the event was release. i.e. the user lifted the finger from the screen or released the mouse button.
+ - `onTapCancel`: Called when the event was cancelled, this can happen for several reasons, one of the most common reason is when the event has changed into another type, like for example the user started to move the finger/mouse and the touch event now turned into a pan/drag type. Usually we can just threat this event as the same as `onTapUp`.
+
+Now that we have a better understanding of `TapDetector` and the events that we will need to handle, lets implement it on the game:
+
+```dart
+// We need to add our `TapDetector` mixin here
+class MyGame extends Game with TapDetector {
+  // Variables declaration, onLoad and render methods omited...
+
+  @override
+  void onTapDown(TapDownDetails details) {
+    // On tap down we need to check if the event ocurred on the
+    // button area. There are several ways of doing it, for this
+    // tutorial we do that by transforming ours position and size
+    // vectors into a dart:ui Rect by using the `&` operator, and
+    // with that rect we can use its `contains` method which checks
+    // if a point (Offset) is inside that rect
+    final buttonArea = buttonPosition & buttonSize;
+
+    if (buttonArea.contains(details.localPosition)) {
+      isPressed = true;
+    }
```

Comment:
we should also probably mention that, for simplicity, we are not properly handling multitap here.

and since we are not anyway, we can simplify this code a bit:

```suggestion
    isPressed = buttonArea.contains(details.localPosition);
```

Reply:
Fixed